IM_calculation/IM/im_calculation.py

```diff
@@ -364,24 +365,34 @@ def calculate_IESD(
     comps_to_calculate,
     z=0.05,  # damping ratio
     alpha=0.05,  # strain hardening ratios
-    dy=0.025,  # strain hardening ratios
+    dy=0.1765,
     dt=0.005,  # analysis time step
 ):
+    # Get displacements by Burks_Baker_2013. Has shape (len(periods), nt-1, len(comps))
+    displacements = intensity_measures.get_SDI_nd(
+        accelerations, im_options[im], waveform_acc.NT, DT, z, alpha, dy, dt
+    )
+
+    # Calculate the maximums of the basic components and pass this to array_to_dict which calculates geom too
+    # Store the SDI im values in the format dict(component: List(im_values))
+    # Where the im_values in the component dictionaries correspond to the periods in the periods list
+    sdi_values = array_to_dict(
+        np.max(np.abs(displacements), axis=1),
+        comps_to_calculate,
+        im,
+        comps_to_store,
+    )
+
+    if check_rotd(comps_to_store):
+        # Only run if any of the given components are selected (Non empty intersection)
+        sdi_values.update(calculate_rotd(displacements, comps_to_store))
 
-    acc_values = array_to_dict(accelerations, comps_to_calculate, im, comps_to_store)
     for comp in comps_to_store:
-        if comp.str_value in acc_values:
-            Sd = Bilinear_Newmark_withTH(
-                np.array(im_options[im]),
-                z,
-                dy,
-                alpha,
-                acc_values[comp.str_value],
-                waveform_acc.DT,
-                dt,
-            )
+        if comp.str_value in sdi_values:
             for i, val in enumerate(im_options[im]):
-                result[(station_name, comp.str_value)][f"{im}_{str(val)}"] = Sd[i]
+                result[(station_name, comp.str_value)][f"{im}_{str(val)}"] = (
+                    sdi_values[comp.str_value][i] * 100
```

Comment:
Why is this being multiplied by 100?

Reply:
The original Burks and Baker wants acceleration in m/s^2  Then displacement wants to be stored in cm.

Reply:
Cool, should put in a comment saying so

Reply:
yeah, the comment was pushed to the next line by black...

Reply:
added more comments.